yamllint/rules/comments.py

```diff
@@ -105,7 +103,7 @@ def check(conf, comment):
             if (conf['ignore-shebangs'] and
                     comment.line_no == 1 and
                     comment.column_no == 1 and
-                    re.match(r'^!\S', comment.buffer[text_start:])):
+                    comment.buffer[text_start:].startswith('!')):
```

Comment:
`comment.buffer[text_start:]` creates a new buffer, so this line is probably not very efficient (especially if the YAML file is large).

Moreover we just need the first char so `startswith()` is not that useful. I propose the following:
```suggestion
                    comment.buffer[text_start] == '!'):
```

Reply:
Done. Thanks for your time.